sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py

```diff
@@ -154,37 +165,146 @@ class ShareFileClient(AsyncStorageAccountHostsMixin, ShareFileClientBase):
         The hostname of the secondary endpoint.
     :keyword int max_range_size: The maximum range size used for a file upload. Defaults to 4*1024*1024.
     :keyword str audience: The audience to use when requesting tokens for Azure Active Directory
-        authentication. Only has an effect when credential is of type TokenCredential. The value could be
+        authentication. Only has an effect when credential is of type Async TokenCredential. The value could be
```

Comment:
nit: no space between Async and TokenCredential since you are referring to the literal `AsyncTokenCredential` type